apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -28,54 +26,29 @@ class GoogleClassroomShareButton extends React.Component {
     courseid: PropTypes.number
   };
 
+  /*
+   * The button doesn't render immediately, so we use its resize event to
+   * detect when it's rendered and wait to display the label until then.
+   */
   constructor(props) {
     super(props);
-    this.loadApi = this.loadApi.bind(this);
-    this.waitForGapi = this.waitForGapi.bind(this);
-    this.renderButton = this.renderButton.bind(this);
-  }
-
-  componentDidMount() {
-    if (this.gapiReady()) {
-      this.renderButton();
-    } else {
-      this.loadApi();
-    }
+    this.onButtonResize = this.onButtonResize.bind(this);
+    this.resizeObserver = new ResizeObserver(this.onButtonResize);
```

Comment:
Rather than watch the HTML element, can you use the redux `canShowGoogleShareButton`

Reply:
`canShowGoogleShareButton` means the api has loaded (and the section/user requirements are met), but per the comment above, the actual rendering of the button is not instantaneous, so i'm using the resize event to detect when it actually appears onscreen to synchronize the showing of the label.